apps/starknet_app/starknet_poseidon.h

```diff
@@ -0,0 +1,128 @@
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include <error.pb.h>
+#include <starknet/sign_txn.pb.h>
+#include <stdint.h>
+
+#include "coin_utils.h"
+#include "f251.h"
+#include "mini-gmp-helpers.h"
+#include "poseidon.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+#define DATA_AVAILABILITY_MODE_BITS 32    // 32 bits for data availability mode
+#define MAX_AMOUNT_BITS 64                // 64 bits for max_amount
+#define MAX_PRICE_PER_UNIT_BITS 128       // 128 bits for max_price_per_unit
+#define RESOURCE_VALUE_OFFSET                                                  \
+  (MAX_AMOUNT_BITS + MAX_PRICE_PER_UNIT_BITS)    // Combined offset
+#define L1_GAS_NAME "4c315f474153"    // The constant value for L1_GAS_NAME
+#define L2_GAS_NAME "4c325f474153"
+#define INVOKE_TXN_PREFIX                                                      \
+  { 0x69, 0x6e, 0x76, 0x6f, 0x6b, 0x65 }    // 0x696e766f6b65; 'INKVOKE'
+#define DEPLOY_ACCOUNT_PREFIX                                                  \
+  {                                                                            \
+    0x64, 0x65, 0x70, 0x6c, 0x6f, 0x79, 0x5f, 0x61, 0x63, 0x63, 0x6f, 0x75,    \
+        0x6e, 0x74                                                             \
+  }    // 0x6465706c6f795f6163636f756e74 'DEPLOY_ACCOUNT'
```

Comment:
Are all of these related to poseidon hash? if not, they shouldn't be in this file

Reply:
yes, releated to starkent context of poseidon hash, all values are used only in `starknet_poseidon.c`, so moved those private macro section in `starknet_poseidon.c`